src/game/market.rs

```diff
@@ -242,9 +242,24 @@ pub fn extend_order(order_id: &str, add_amount: u32) -> ReturnCode {
 
 /// Get all orders from the market
 ///
-/// Contrary to the JS version, filtering should be done afterwards.
-pub fn get_all_orders() -> Vec<Order> {
-    js_unwrap!(Game.market.getAllOrders())
+/// Full filtering support is not available, but filtering by resource type
+/// is available and will reduce the CPU cost compared to getting all orders
+pub fn get_all_orders(resource: Option<MarketResourceType>) -> Vec<Order> {
+    match resource {
+        Some(resource_type) => match resource_type {
```

Comment:
I think we could avoid multiple js_unwraps in this case if we used `as u32` in each branch, and stored the result in a variable? Not sure if that would actually compile any differently, though.